examples/angular/src/app/lib/layout/container/container.component.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnChanges,
+  computed,
+  inject,
+  signal,
+} from '@angular/core';
+import { AsyncPipe, NgComponentOutlet } from '@angular/common';
+
+import { getContainersData } from '../../utils';
+import {
+  PageContextService,
+} from '../../services/dotcms-context/page-context.service';
+import { NoComponentComponent } from '../../components/no-component/no-component.component';
+import { DotCMSContainer, DotCMSContentlet, DynamicComponentEntity } from '../../models';
+
+interface DotContainer {
+  acceptTypes: string;
+  identifier: string;
+  maxContentlets: number;
+  uuid: string;
+  variantId?: string;
+}
+
+const EMPTY_CONTAINER_EDIT_MODE_STYLES = {
+  width: '100%',
+  backgroundColor: '#ECF0FD',
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  color: '#030E32',
+  height: '10rem',
+};
+
+@Component({
+  selector: 'dotcms-container',
+  standalone: true,
+  imports: [AsyncPipe, NgComponentOutlet, NoComponentComponent],
+  templateUrl: './container.component.html',
+  styleUrl: './container.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ContainerComponent implements OnChanges {
+    @Input({ required: true }) container!: DotCMSContainer;
+
+    private readonly pageContextService: PageContextService = inject(PageContextService);
+    protected readonly emptyContainerStyles: Record<string, string> = EMPTY_CONTAINER_EDIT_MODE_STYLES;
+    protected readonly NoComponentComponent = NoComponentComponent;
+    protected readonly isInsideEditor = signal<boolean>(false);
+
+    protected componentsMap!: Record<string, DynamicComponentEntity>;
+    protected contentlets = signal<DotCMSContentlet[]>([]);
+    protected dotContainer = signal<DotContainer | null>(null);
+    protected dotContainerAsString = computed(() =>
+        JSON.stringify(this.dotContainer())
+    );
+
+    ngOnChanges() {
```

Comment:
Why this is not `ngOnInit`?

Reply:
Catch all Strategy for routing does not trigger the `ngOnInit` when you change route. [[More information](https://stackoverflow.com/questions/40951695/ngoninit-not-firing-one-change-between-children-routes)].

Since we use the `container` input to get the `contentlets` directly in the **component**, not the **template**, we need to use the `ngOnChanges`.